python/test/unit/runtime/test_autotuner.py

```diff
@@ -20,3 +20,20 @@ def _kernel(dst, src, N, BLOCK_SIZE: tl.constexpr):
     grid = lambda META: (triton.cdiv(N, META['BLOCK_SIZE']),)
     _kernel[grid](dst, src, N)
     _kernel[grid](dst=dst, src=src, N=N)
+
+
+def test_restore():
+    N = 1024
+    src = torch.zeros(N, device='cuda')
+
+    configs = [triton.Config(kwargs={'BLOCK_SIZE': 32}), triton.Config(kwargs={'BLOCK_SIZE': 128})]
+
+    @triton.autotune(configs=configs, key=['N'], restore_value=['src'])
+    @triton.jit
+    def _kernel(src, N, BLOCK_SIZE: tl.constexpr):
+        offsets = tl.program_id(0) * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+        x = tl.load(src + offsets, mask=offsets < N) + 1
```

Comment:
I believe there has to be a debug_barrier between load and store.

Otherwise, thread 0 may store 1 to loc 0, and then thread 64 may load 1 and store 2 to loc 0, when BLOCK_SIZE=32.

Reply:
Ah, I remembered it wrong. The duplicate store problem should have been fixed long ago